Pass host to onInjection Hook + Add docs

iMostfa opened the pull request:
<img width="314" alt="image" src="https://github.com/krzysztofzablocki/Inject/assets/2325884/5998416f-34c4-4570-be1d-fe432a34f822">

Hello, in This PR i add documentation for the small feature introduced in  
#91
where it describes when it might be useful, with an example, thanks for the great project <3

krzysztofzablocki commented:
@iMostfa you are not really passing any argument to the hook so I don't think that works this way? also you might want to change the prototype to pass in the host so you don't need to cast in the closure itself ?

iMostfa commented:
@krzysztofzablocki oh, my previous example wasn't the best, in the example i was capturing the `presenter` from` self/the world of caller`


<img width="840" alt="Screenshot 2024-04-08 at 6 01 15 PM" src="https://github.com/krzysztofzablocki/Inject/assets/2325884/8d7486e3-4b38-4189-9a29-3174d1f8fdc2">

- i updated the hook to pass the host
- i updated the comment in the code

krzysztofzablocki commented:
🙇
